### PR TITLE
Creating Sparse Matrix from Pdarrays

### PR DESCRIPTION
**Pull request by ShreyasKhandekar**
Adds functionality to create sparse matrices from pdarrays.

Adds testing.
Next steps would be to optimize this algorithm
Tested on multilocale.

### CODE REVIEW

**Comment by ShreyasKhandekar** [comment]
The `mypy` failures seem to be originating elsewhere, not related to my changes:

```console
Run python3 -m mypy arkouda
arkouda/util.py:614: note: By default the bodies of untyped functions are not checked, consider using --check-untyped-defs  [annotation-unchecked]
arkouda/segarray.py:71: note: By default the bodies of untyped functions are not checked, consider using --check-untyped-defs  [annotation-unchecked]
arkouda/pdarraycreation.py:926: error: Incompatible return value type (got "Union[Any, pdarray, Strings]", expected "pdarray")  [return-value]
arkouda/groupbyclass.py:153: error: Incompatible return value type (got "tuple[Union[Any, Categorical], Any, Any, Any, Union[Any, tuple[Union[Any, Categorical], Any, Any, Any]]]", expected "Union[Union[Union[pdarray, Strings, Categorical], Sequence[Union[pdarray, Strings, Categorical]]], tuple[Union[Union[pdarray, Strings, Categorical], Sequence[Union[pdarray, Strings, Categorical]]], pdarray, pdarray, int]]")  [return-value]
Found 2 errors in 2 files (checked 82 source files)
Error: Process completed with exit code 1.
```

**Review by bmcdonald3 — CHANGES_REQUESTED** [review]

File: arkouda/sparrayclass.py [hunk 1]
```diff
@@ -115,7 +117,6 @@ def __str__(self):  # This won't work out of the box for sparrays need to add th
     [array([1 1 1 ... 100 100 100]), array([17 21 29 ... 75 77 85]), array([0 0 0 ... 0 0 0])]
     """
 
-    @typechecked
     def to_pdarray(self):
```
Comment (bmcdonald3) [inline arkouda/sparrayclass.py:120]: I think the failures for mypy are related to something in this file. Basically, I think you are creating a pdarray type that isn't supported in the pdarray creation functions, but that is just with a quick glance.

Comment (bmcdonald3) [reply arkouda/sparrayclass.py:120]: Also, FWIW, you can run `mypy` locally to see if you are seeing the same thing

Comment (ShreyasKhandekar) [reply arkouda/sparrayclass.py:120]: Running `python3 -m mypy arkouda` locally gives me:

```
python3 -m mypy arkouda
Success: no issues found in 82 source files
```

Comment (ShreyasKhandekar) [reply arkouda/sparrayclass.py]: In any case, I've re-added the `@typechecked` attribute as well as a new type annotation to the `to_pdarray` method.

**Comment by ShreyasKhandekar** [comment]
I have run the `mypy` checks locally on my mac and on a cluster and I don't see errors. I am confused by why the CI is failing on Github actions.

**Comment by ShreyasKhandekar** [comment]
I've rebased with main now that https://github.com/Bears-R-Us/arkouda/pull/3843 is merged, so I expect the checks to pass this time around.

**Review by bmcdonald3 — APPROVED** [review]

**Pull request merged**